src/builder.ts

```diff
@@ -83,10 +83,21 @@ export default class Builder {
 		// Connect the input stream to the rw stream
 		dup.setWritable(inputStream);
 
-		Bluebird.resolve(this.docker.buildImage(inputStream, buildOpts))
-			.then((res: NodeJS.ReadWriteStream) => {
-				let errored = false;
-				const outputStream = res
+		let streamError: Error;
+		const failBuild = _.once((err: Error) => {
+			streamError = err;
+			dup.destroy(err);
+			return this.callHook(hooks, 'buildFailure', handler, err, this.layers);
```

Comment:
is the `return` needed?

Reply:
Not strictly needed, no, but by returning hook promise, that promise is awaited in the `.catch(failBuild)` call, which makes it less prone to unhandled rejection errors in case someone later adds a `.then` after the `.catch`. Well, in any case, it should at least be harmless.